chipsec/chipset.py

```diff
@@ -835,6 +835,33 @@ def read_register_all(self, reg_name, cpu_thread=0):
             values.append( self.read_register( reg_name, cpu_thread, index) )
         return values
 
+    def read_all_register_instances(self, reg_name):
+        values = []
+        reg = self.get_register_def(reg_name)
```

Comment:
Not sure I understand the comment, get_register_def should raise an Exception if the register name does not exist.

Reply:
`get_register_def()` does not check if `reg_name` is defined before trying to fetch it.  Should not rely on error to catch if a `reg_name` is (not) defined for a platform.  This can be checked and return gracefully if not; eg, `return []`.